fix(ci): backfill-generate-matrix.sh writes multi-line JSON to GITHUB_OUTPUT

## Summary

The `backfill-docker-tags.yml` workflow fails on all 6 batches because `backfill-generate-matrix.sh` writes multi-line JSON to `GITHUB_OUTPUT`, which GitHub Actions rejects as an invalid file command.

## Problem

Line 56 of `scripts/ci/backfill-generate-matrix.sh`:

```bash
echo "tags=${json}" >>"$GITHUB_OUTPUT"
```

The `$json` variable contains a pretty-printed JSON array (multi-line). GitHub Actions' `GITHUB_OUTPUT` file command requires either single-line values or heredoc delimiters for multi-line content. The `{` on line 2 of the JSON is interpreted as an invalid file command:

```
##[error]Unable to process file command 'output' successfully.
##[error]Invalid format '  {'
```

### Evidence

All 6 backfill batches fail at the "Generate tag matrix" step:
- Batch 1: https://github.com/lgtm-hq/py-lintro/actions/runs/23213817380
- Batch 2: https://github.com/lgtm-hq/py-lintro/actions/runs/23213818839
- Batch 3: https://github.com/lgtm-hq/py-lintro/actions/runs/23213820137
- Batch 4: https://github.com/lgtm-hq/py-lintro/actions/runs/23213821806
- Batch 5: https://github.com/lgtm-hq/py-lintro/actions/runs/23213823286
- Batch 6: https://github.com/lgtm-hq/py-lintro/actions/runs/23213824882

## Proposed Solution

Compact the JSON to a single line with `jq -c` before writing to `GITHUB_OUTPUT`:

```bash
echo "tags=$(echo "$json" | jq -c '.')" >>"$GITHUB_OUTPUT"
```

## Files to Modify

- `scripts/ci/backfill-generate-matrix.sh` — line 56, compact JSON before writing

## References

- Parent issue: #730
- [GitHub docs: GITHUB_OUTPUT](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-output-parameter)

--- a/scripts/ci/backfill-generate-matrix.sh
+++ b/scripts/ci/backfill-generate-matrix.sh
@@ -50,7 +50,7 @@ for ((i = start; i < end; i++)); do
 	sha=$(git rev-parse "$tag^{}")
 	objects+=("$(jq -n --arg tag "$tag" --arg sha "$sha" '{tag:$tag,sha:$sha}')")
 done
-json=$(printf '%s\n' "${objects[@]}" | jq -s '.')
+json=$(printf '%s\n' "${objects[@]}" | jq -sc '.')
 count=${#objects[@]}
 
 echo "tags=${json}" >>"$GITHUB_OUTPUT"

